src/main/java/org/openlmis/requisition/errorhandling/ValidationResult.java

```diff
@@ -0,0 +1,166 @@
+    ValidationResult result = new ValidationResult();
+    result.addError(new Message(messageKey, msgParameters), VALIDATION);
+    return result;
+  }
+
+  /**
+   * Creates a ValidationResult for a check that failed due to a reference to
+   * an instance that did not exist.
+   *
+   * @param messageKey a message key to include in the error or exception
+   * @param msgParameters parameters for the given message
+   * @return a ValidationResult instance containing not found error
+   */
+  public static ValidationResult notFound(String messageKey, Object... msgParameters) {
+    ValidationResult result = new ValidationResult();
+    result.addError(new Message(messageKey, msgParameters), NOT_FOUND);
+    return result;
+  }
+
+  /**
+   * Creates a ValidationResult for a check that failed due to a user not having
+   * permissions to access the resource.
+   *
+   * @param messageKey a message key to include in the error or exception
+   * @param msgParameters parameters for the given message
+   * @return a ValidationResult instance containing permission error
+   */
+  public static ValidationResult noPermission(String messageKey, Object... msgParameters) {
+    ValidationResult result = new ValidationResult();
+    result.addError(new Message(messageKey, msgParameters), NO_PERMISSION);
+    return result;
+  }
+
+  /**
+   * Creates a ValidationResult for a check that failed due to a conflict.
+   *
+   * @param messageKey a message key to include in the error or exception
+   * @param msgParameters parameters for the given message
+   * @return a ValidationResult instance containing conflict error
+   */
+  public static ValidationResult conflict(String messageKey, Object... msgParameters) {
+    ValidationResult result = new ValidationResult();
+    result.addError(new Message(messageKey, msgParameters), CONFLICT);
+    return result;
+  }
+
+  /**
+   * Creates a ValidationResult for a check that succeeded.
+
+   * @return a ValidationResult instance representing successful validation
+   */
+  public static ValidationResult success() {
+    return new ValidationResult();
+  }
+
+  /**
+   * Adds a new error to this validation result.
+   *
+   * @param message a message key to include in the error or exception
+   * @param type a type of an error
+   */
+  public void addError(Message message, FailureType type) {
+    this.errors.add(new ValidationFailure(message, type));
+  }
+
+  /**
+   * Checks if there are any errors in this validation result.
+   * @return true if there are any errors; false otherwise
+   */
+  public boolean hasErrors() {
+    return !isSuccess();
+  }
+
+  /**
+   * Checks if there are no errors in this validation result.
+   * @return true if there are no errors; false otherwise
+   */
+  public boolean isSuccess() {
+    return errors.isEmpty();
+  }
+
+  /**
+   * Retrieves all errors of this validation result.
+   * @return a list of errors
+   */
+  public List<ValidationFailure> gerErrors() {
```

Comment:
What will be the use of this method in future? multiple errors in http response?